echoregions/regions2d/regions2d.py

```diff
@@ -656,49 +616,48 @@ def transect_mask(
+        warning_messages = []
 
-        # Check if resume_str followed by break_str/end_str.
-        resume_transect_rows = transect_df[transect_df["transect_type"] == resume_str]
-        resume_transect_type_next_list = list(resume_transect_rows["transect_type_next"].values)
-        for transect_type_next in resume_transect_type_next_list:
-            if transect_type_next not in [break_str, end_str]:
-                raise ValueError(
-                    f"Transect resume string is followed by invalid value "
-                    f"{transect_type_next}. Must be followed by either "
-                    f"{break_str} or {end_str}."
-                )
+        # Set transect_type_next values to be empty strings if they are NAs.
+        transect_df["transect_type_next"] = transect_df.apply(
+            lambda x: "" if isna(x["transect_type_next"]) else x["transect_type_next"],
+            axis=1,
+        )
 
-        # Check if end_str followed by start_str or if NA.
-        end_transect_rows = transect_df[transect_df["transect_type"] == end_str]
-        end_transect_type_next_list = list(end_transect_rows["transect_type_next"].values)
-        for transect_type_next in end_transect_type_next_list:
-            # If this value is not NA, check if it is start_str.
-            if not isna(transect_type_next):
-                if transect_type_next != start_str:
-                    raise ValueError(
-                        f"Transect end string is followed by invalid value "
-                        f"{transect_type_next}. Must be followed by {start_str}."
-                    )
+        # Ensure correct sequence of transect types occur. If they do not, append to warning_messages.
+        for index, row in transect_df.iterrows():
+            transect_type = row["transect_type"]
+            transect_type_next = row["transect_type_next"]
+            type_next_warning_message = (
+                f"Error in row {index + 2}: Transect string {transect_type} is followed by "
+                f"invalid value {transect_type_next}. Must be followed by "
+                f"{transect_next_allowable[transect_type]}"
+            )
+            # Check for correct transect_type_next values
+            if transect_type_next not in transect_next_allowable[transect_type]:
+                warning_messages.append(type_next_warning_message)
+
+        # Checking the maximum width of a transect log region bbox.
+        # If over a minute, throw an error.
+        max_time = (transect_df["region_bbox_right"] - transect_df["region_bbox_left"]).max()
+        max_time_minutes = max_time.total_seconds() / 60
+        if max_time_minutes > bbox_distance_threshold:
+            warning_messages.append(
+                f"Maximum width in time of transect log region bboxs is "
+                f"too large i.e. over {bbox_distance_threshold} minute(s). "
+                f"The maximum width is: {max_time_minutes}.",
+            )
```

Comment:
Why is this checking only the `max_time`/`max_time_minutes` instead of showing which entry it has such a problem? Wouldn't it be more efficient to have that if the goal is to correct the input dataframe?

Reply:
Oh that is true. I can make this change.